openlibrary/plugins/admin/code.py

```diff
@@ -54,6 +55,80 @@ def register_admin_page(path, cls, label=None, visible=True, librarians=False):
     admin_tasks.append(t)
 
 
+def revert_all_user_edits(account: Account) -> tuple[int, int]:
+    """
+    :return: tuple of (number of edits reverted, number of documents deleted)
+    """
+    i = 0
+    edit_count = 0
+    stop = False
+    keys_to_delete = set()
+    while not stop:
+        changes = account.get_recentchanges(limit=100, offset=100 * i)
+        added_records: list[list[dict]] = [
+            c.changes for c in changes if c.kind == 'add-book'
+        ]
+        flattened_records: list[dict] = sum(added_records, [])
+        keys_to_delete |= {r['key'] for r in flattened_records}
+
+        keys_to_revert: dict[str, list[int]] = {
+            item.key: [] for change in changes for item in change.changes
+        }
```

Comment:
```suggestion
        keys_to_revert: dict[str, list[int]] = defaultdict(list)
```

+ Add it to the import

Reply:
@benbdeitch we do need the mapping to the empty list! Which is correct!